src/GraphQL.Client/GraphQLHttpClientOptions.cs

```diff
@@ -67,5 +67,11 @@ public class GraphQLHttpClientOptions
         /// See https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md#gql_connection_init.
         /// </summary>
         public Func<GraphQLHttpClientOptions, object?> ConfigureWebSocketConnectionInitPayload { get; set; } = options => null;
+
+
+        /// <summary>
+        /// Add the assembly name as UserAgent request header if none is set.
+        /// </summary>
+        public bool AddDefaultUserAgentRequestHeader { get; set; } = true;
```

Comment:
I would suggest to not touch `DefaultRequestHeaders` at all:

1. Change new option to something like 
```
public Func<GraphQLClientHttpOptions, ProductInfoHeaderValue> DefaultUserAgentRequestHeader { get; set; } = _ => new ProductInfoHeaderValue(GetType().Assembly.GetName().Name, GetType().Assembly.GetName().Version.ToString())
```

2. Remove
```
 if (!HttpClient.DefaultRequestHeaders.UserAgent.Any())
     HttpClient.DefaultRequestHeaders.UserAgent.Add(ProductInfoHeaderValue(GetType().Assembly.GetName().Name, GetType().Assembly.GetName().Version.ToString()));
```

3. Add 
```
            var userAgent = options.DefaultUserAgentRequestHeader(options);
            if (userAgent != null)
                message.Headers.UserAgent.Add(userAgent);
```

into `GraphQLHttpRequest.ToHttpRequestMessage`.